cairo/tests/utils/args_gen.py

```diff
@@ -1,3 +1,54 @@
+"""
+Cairo Type System - Argument Generation
+
+This module handles the generation of Cairo memory values from Python types.
+It is a core component of the type system that allows seamless conversion of Python
+values into the appropriate Cairo memory layout.
+
+Type System Patterns:
+
+1. Type Wrapping Pattern:
+   - All complex types are wrapped in a pointer-based structure
+   - Example: `Bytes { value: BytesStruct* }` where BytesStruct contains actual data
+   - This ensures all complex types have a consistent size of 1 pointer
+
+2. None Value Pattern:
+   - None is represented by a null pointer (pointer to 0)
+   - For simple types (size 1), we use direct pointers (e.g., Uint*)
+   - This optimizes memory by storing [value] instead of [ptr_value, value]
```

Comment:
still wondering if this is a good decision, leaning toward no as we don't want to depend on the felt size, and for example U64 was one felt and will become 3

Reply:
I think this could be part of a much bigger work regarding STWO optimizations, as it basically means we need to get rid of all starkware-provided core library, right ? As all assumptions of computations not overflowing the felt size will not hold anymore